pages/desktop/consumer_pages/base.py

```diff
@@ -51,25 +47,16 @@ def go_to_debug_page(self):
         return Debug(self.testsetup)
 
     def set_region(self, region):
-
         debug_page = self.go_to_debug_page()
         debug_page.select_region(region)
-
-        self.wait_notification_box_visible()
-        if not self.notification_visible:
-            raise Exception('Unable to change region to %s. Notification not displayed'
-                            % (region))
-
-        self.wait_notification_box_not_visible()
+        self.wait_for_notification()
```

Comment:
> you have changed some other calls which were not waiting for a specific message to do so

Are you sure? I was careful not to. It can be done in a later refactor though.